warehouse/core/src/main/java/datawave/ingest/mapreduce/job/RFileSplit.java

```diff
@@ -0,0 +1,75 @@
+package datawave.ingest.mapreduce.job;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.lib.input.FileSplit;
+
+// For a file with N blocks
+// startBlock 0 - beginning of the file to the first block
+// startBlock N - last block to the end
+public class RFileSplit extends FileSplit {
+    private long startBlock;
+    private long numBlocks;
+    private Key top;
+
+    public RFileSplit() {
+        top = new Key();
+    }
+
+    public RFileSplit(Path path, long fileStart, long fileLength, String[] hosts, int startBlock, int numBlocks, Key top) {
+        super(path, fileStart, fileLength, hosts);
+        this.startBlock = startBlock;
+        this.numBlocks = numBlocks;
+        this.top = top;
+    }
+
+    public long getStartBlock() {
+        return startBlock;
+    }
+
+    public Key getTopKey() {
+        return top;
+    }
+
+    /**
+     * Create a range which is start key inclusive and end key exclusive
+     *
+     * @param start
+     * @param end
+     * @return
+     */
+    public Range getSeekRange(Key start, Key end) {
+        return new Range(start, true, end, false);
+    }
+
+    @Override
+    public long getLength() {
+        return numBlocks;
```

Comment:
The API mentions this is the number of bytes that will be read from the split. We're converting it to the a number of blocks - which is fine probably as the length relative to other splits will stay similar. Was there a reason to override this vs just leaving it as the default of `fileLength`?

Reply:
I see down below where we use the `getLength()` method to retrieve the number of index blocks. We could create a new method for `getNumBlocks()` like you did for the `startBlock`.

Reply:
your right, there isn't a good reason to break the contract, will update